### PR TITLE
Feature/follow

### PR DESCRIPTION
**Pull request by active-sludge**
A relation table of the following model is created.
Users can follow/unfollow each other.
Followers and followings count is printed on the profile page.
Followers and followings list is dynamically listed on modal windows.
Usernames are linked to their profile.

### CODE REVIEW

**Comment by active-sludge** [comment]
Follow button on the followers/following list is hidden.
That feature will be separated from this feature. 
The issue is created #132 .

File: Project/Wikode/wikodeApp/static/wikodeApp/js/profilePage.js [hunk 1]
```diff
@@ -40,16 +33,21 @@ const showModal = function(header, list) {
         let tdItem = document.createElement("td")
 
         let itemAnchor = document.createElement("a");
-        itemAnchor.innerHTML = item
-        itemAnchor.href = "#"
+        let first_name = item[1];
+        let last_name = item[2];
```
Comment (jajimentol) [inline Project/Wikode/wikodeApp/static/wikodeApp/js/profilePage.js:37]: index range leri kontrol edelim

File: Project/Wikode/wikodeApp/views.py [hunk 2]
```diff
@@ -281,7 +282,101 @@ def getArticles(request):
...
+    }
+
+    return render(request, 'wikodeApp/profilePage.html', context)
+
+
+## Renders the profilePage.html with the clicked user's id information as pk
+## Navigates to /profile/# url.
+@login_required
+def getProfilePageOfOtherfUser(request, pk):
+    ## TODO
```
Comment (jajimentol) [inline Project/Wikode/wikodeApp/views.py:307]: typo

File: Project/Wikode/wikodeApp/static/wikodeApp/js/profilePage.js (hunk 1 above)
Comment (active-sludge) [reply Project/Wikode/wikodeApp/static/wikodeApp/js/profilePage.js:37]: Eklendi.

File: Project/Wikode/wikodeApp/views.py [hunk 3]
```diff
@@ -281,7 +282,101 @@ def getArticles(request):
...
+
+
+    ## Return Follow/Unfollow button appearance is determined by is_followed value.
+    ## If True don't show Follow button, show Unfollow instead.
+    return redirect('wikodeApp:getProfilePageOfOtherfUser', pk)
+
+
+## Create a followee list from Follow relation that the follower id is other_user.id
+## List example: [1, 'Can', 'Dayan']
+def getFolloweeList(user):
```
Comment (semihsolmaz) [inline Project/Wikode/wikodeApp/views.py:357]: move this to utils under a approoriate user manager file

File: Project/Wikode/wikodeApp/views.py [hunk 4]
```diff
@@ -281,7 +282,101 @@ def getArticles(request):
...
+                .filter(follower_id=user.id)
+                .values_list('followee_id', 'followee__first_name', 'followee__last_name')
+        )
+    )
+    return followee_list
+
+
+## Create a follower list from Follow relation that the followee id is other_user.id
+## List example: [1, 'Can', 'Dayan']
+def getFollowerList(user):
```
Comment (semihsolmaz) [inline Project/Wikode/wikodeApp/views.py:371]: move this to utils under a appropriate user manager file

File: Project/Wikode/wikodeApp/views.py (hunk 2 above)
Comment (active-sludge) [reply Project/Wikode/wikodeApp/views.py:307]: changed to getProfilePageOfOtherUser

**Review by semihsolmaz — CHANGES_REQUESTED** [review]

File: Project/Wikode/wikodeApp/views.py (hunk 3 above)
Comment (active-sludge) [reply Project/Wikode/wikodeApp/views.py:357]: followManager created inside utils. Functions are moved there.

File: Project/Wikode/wikodeApp/views.py (hunk 4 above)
Comment (active-sludge) [reply Project/Wikode/wikodeApp/views.py:371]: followManager created inside utils. Functions are moved there.

**Review by semihsolmaz — APPROVED** [review]

**Pull request merged**